pkg/api/v20170131/types.go

```diff
@@ -28,7 +28,7 @@ type ContainerService struct {
 
 // Properties represents the ACS cluster definition
 type Properties struct {
-	ProvisioningState       *ProvisioningState       `json:"provisioningState,omitempty"`
+	ProvisioningState       ProvisioningState        `json:"provisioningState,omitempty"`
```

Comment:
In Properties struct, it is also not a pointer. And in other versions, it is not a pointer also. It looks to me a typo here.